debezium-server-pulsar/src/main/java/io/debezium/server/pulsar/PulsarChangeConsumer.java

```diff
@@ -152,4 +175,52 @@ public void handleBatch(List<ChangeEvent<Object, Object>> records, RecordCommitt
         }
         committer.markBatchFinished();
     }
+
+    @SuppressWarnings("unchecked")
+    private void handleBatchAsync(List<ChangeEvent<Object, Object>> records, RecordCommitter<ChangeEvent<Object, Object>> committer) throws InterruptedException {
+        List<CompletableFuture<MessageId>> futures = new ArrayList<>();
+
+        for (ChangeEvent<Object, Object> record : records) {
+            final TypedMessageBuilder message = buildMessage(record);
+
+            final ChangeEvent<Object, Object> currentRecord = record;
+
+            CompletableFuture<MessageId> future = message
+                    .sendAsync()
+                    .thenAccept(messageId -> {
+                        LOGGER.trace("Sent message with id: {}", messageId);
+                        try {
+                            committer.markProcessed(currentRecord);
+                        }
+                        catch (InterruptedException e) {
+                            throw new DebeziumException(e);
+                        }
+                    })
+                    .exceptionally(ex -> {
+                        Throwable exception = (Throwable) ex;
+                        LOGGER.error("Failed to send record to {}:", record.destination(), exception);
```

Comment:
It will be a message-based error log.
For Pulsar, if the first message is failed to publish, all the subsequent messages will also be failed.
I think we can change the log level to trace here and add an error log between line:214 and line:215.
To make sure it will not be noisy.

Reply:
Done.